server.py

```diff
@@ -109,8 +109,33 @@ async def restart_control_server():
         )
     config.register_callback("CONTROL_SERVER_PORT", restart_control_server)
 
-    await instance.listen(("", 8001))
-    await instance.listen(("", 8002), SimpleJsonProtocol)
+    PROTO_CLASSES = {
+        QDataStreamProtocol.__name__: QDataStreamProtocol,
+        SimpleJsonProtocol.__name__: SimpleJsonProtocol
+    }
+    for cfg in config.LISTEN:
+        try:
+            host = cfg.get("address", "")
+            port = cfg.get("port", 0)
+            proto_class_name = cfg.get("protocol", QDataStreamProtocol.__name__)
```

Comment:
Not sure if these are the best defaults. Using `0` for port makes it listen on a random open high port.

Reply:
Why have defaults? Wouldn't it be better to have it fail fast with a misconfiguration?

Reply:
We could do it that way as well